identity/src/commonMain/kotlin/com/android/identity/nfc/NdefRecord.kt

```diff
@@ -87,19 +91,21 @@ data class NdefRecord(
         if (isShortRecord) flags = flags or FLAG_SR
         if (idLengthPresent) flags = flags or FLAG_IL
 
-        buf.writeByte(flags.toByte())
-        buf.writeByte(type.size.toByte())
+        bsb.appendUInt8(flags)
+        bsb.appendUInt8(type.size)
         if (isShortRecord) {
-            buf.writeByte(payload.size.toByte())
+            bsb.appendUInt8(payload.size)
         } else {
-            buf.writeInt(payload.size)
+            bsb.appendUInt32(payload.size) // TODO: b/393388370 - is it really 4 byte? or perhaps it's 2 (UInt16)?
         }
         if (idLengthPresent) {
-            buf.writeByte(id.size.toByte())
+            bsb.appendUInt8(id.size)
         }
-        buf.write(type.toByteArray())
-        buf.write(id.toByteArray())
-        buf.write(payload.toByteArray())
+        // TODO: b/393388370 - it looks like a common issue with different ByteArray behaviour for empty data.
+        //  A dedicated appendArray(val:ByteArray) could help.
+        if (type.size > 0) bsb.append(type.toByteArray()) //else bsb.appendUInt8(0)
+        if (id.size > 0) bsb.append(id.toByteArray()) //else bsb.appendUInt8(0)
+        if (payload.size > 0) bsb.append(payload.toByteArray()) //else bsb.appendUInt8(0)
```

Comment:
So there's definitely something fishy here. First of all, there's an [append() method which takes a ByteString](https://github.com/Kotlin/kotlinx-io/blob/master/bytestring/common/src/ByteStringBuilder.kt#L112) which we should use. Second of all, appending an empty ByteArray should definitely be a no-op.

Reply:
Is this still relevant?